package/MDAnalysis/lib/distances.py

```diff
@@ -285,35 +295,42 @@ def self_distance_array(reference, box=None, result=None, backend="serial"):
     ``(N*(N-1)/2)`` is provided in *result* then this preallocated array
     is filled. This can speed up calculations.
 
-    :Arguments:
-        *ref*
-             reference coordinate array with N=len(ref) coordinates
-        *box*
-             cell dimensions (minimum image convention is applied)
-             or None [``None``]
-             Cell dimensions must be in an identical to format to those returned
-             by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`,
-             [lx, ly, lz, alpha, beta, gamma]
-        *result*
-             optional preallocated result array which must have the shape
-             (N*(N-1)/2,) and dtype ``numpy.float64``. Avoids creating
-             the array which saves time when the function is called
-             repeatedly. [``None``]
-        *backend*
-             select the type of acceleration; "serial" is always available. Other
-             possibilities are "OpenMP" (OpenMP).
-    :Returns:
-        *d*
-             N*(N-1)/2 numpy 1D array with the distances dist[i,j] between ref
-             coordinates i and j at position d[k]. Loop through d::
-
-                 for i in range(N):
-                     for j in range(i+1, N):
-                         k += 1
-                         dist[i,j] = d[k]
-
-    .. Note:: This method is slower than it could be because internally we need to
-              make copies of the coordinate arrays.
+    Parameters
+    ----------
+    reference : array
+        Reference coordinate array with ``N=len(ref)`` coordinates.
+    box : array or None
+        Dimensions of the cell; if provided, the minimum image convention is
+        applied. The dimensions must be provided in the same format as returned
+        by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`: ``[lx,
+        ly, lz, alpha, beta, gamma]``.
+    result : array, optional
+        Preallocated result array which must have the shape
+        ``(N*(N-1)/2,)`` and dtype ``numpy.float64``. Avoids creating
+        the array which saves time when the function is called
+        repeatedly. [``None``]
+    backend
+        Select the type of acceleration; "serial" is always available. Other
+        possibilities are "OpenMP" (OpenMP).
+
+    Returns
+    -------
+    d : array
+        ``N*(N-1)/2`` numpy 1D array with the distances dist[i,j] between ref
+        coordinates i and j at position d[k]. Loop through d:
+
+        .. code-block:: python
```

Comment:
Out of curiosity, do we need to markup code as python? I think sphinx's default is python.